src/uwtools/utils/file.py

```diff
@@ -48,6 +50,33 @@ class _FORMAT:
 FORMAT = _FORMAT()
 
 
+class StdinProxy:
+    """
+    Reads stdin once but permits multiple reads of its data.
+    """
+
+    def __init__(self) -> None:
+        self._stdin = sys.stdin.read()
+        self._reset()
+
+    def __getattr__(self, attr: str) -> Any:
+        self._reset()
+        return getattr(self._stringio, attr)
```

Comment:
`__getattr__()` delegates attribute accesses (e.g. `.read()`) to the `StringIO` object referred to by `self._stringio`. `self._reset()` is called first, though, to ensure that the `StringIO` object is reset to its pristine state -- in particular, that it is ready to `.read()` again.

Reply:
Much more effective than what I was hammering away at, we'll need this again for sure